modules/afmongodb/afmongodb-private.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2010-2016 Balabit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef AFMONGODB_PRIVATE_H_
+#define AFMONGODB_PRIVATE_H_
+
+#include "mongoc.h"
+#include "logthrdestdrv.h"
+#include "string-list.h"
+#include "value-pairs/value-pairs.h"
+
+typedef struct _MongoDBDestDriver
+{
+  LogThrDestDriver super;
+
+  /* Shared between main/writer; only read by the writer, never
+   written */
+  gchar *coll;
+  GString *uri_str;
+
+#if SYSLOG_NG_ENABLE_LEGACY_MONGODB_OPTIONS
+  GList *servers;
+  gchar *address;
+  gint port;
+
+  gboolean safe_mode;
+  gchar *user;
+  gchar *password;
+#endif
+
```

Comment:
If we already have a separate layer to format the URL, we might as well really use a separate struct that is embedded here.